src/main/java/com/gitblit/manager/RepositoryManager.java

```diff
@@ -194,10 +194,12 @@ public Date getLastActivityDate() {
 		Date date = null;
 		for (String name : getRepositoryList()) {
 			Repository r = getRepository(name);
-			Date lastChange = JGitUtils.getLastChange(r).when;
-			r.close();
-			if (lastChange != null && (date == null || lastChange.after(date))) {
-				date = lastChange;
+			if(r != null) {
+				Date lastChange = JGitUtils.getLastChange(r).when;
+				r.close();
```

Comment:
Better do `r.close()` in `finally` (so do a `try ... finally` here). Otherwise `r` might leak an open connection if an exception is thrown before (highly unlikely, but not 100% unlikely).